test/validation/api/traffic_mngr/traffic_mngr.c

```diff
@@ -386,35 +385,32 @@ static odp_bool_t approx_eq64(uint64_t val, uint64_t correct)
 static int test_overall_capabilities(void)
 {
 	odp_tm_level_capabilities_t *per_level;
-	odp_tm_capabilities_t        capabilities_array[MAX_CAPABILITIES];
+	odp_tm_capabilities_t        capabilities_array[2];
 	odp_tm_capabilities_t       *cap_ptr;
+	odp_tm_egress_t              egress;
 	uint32_t                     num_records, idx, num_levels, level;
 	int                          rc;
 
-	rc = odp_tm_capabilities(capabilities_array, MAX_CAPABILITIES);
+	odp_tm_egress_init(&egress);
+	egress.egress_kind = ODP_TM_EGRESS_PKT_IO;
+	egress.pktio = xmt_pktio;
+
+	rc = odp_tm_egress_capabilities(&capabilities_array[0], &egress);
 	if (rc < 0) {
 		CU_ASSERT(rc < 0);
 		return -1;
 	}
+	num_records = 1;
 
-	/* Now test the return code (which did not indicate a failure code)
-	 * to make sure that there is at least ONE capabilities record
-	 * returned */
-	if (rc == 0) {
-		CU_ASSERT(rc != 0);
-		return -1;
-	}
-
-	/* Now test the return code to see if there were more capabilities
-	 * records than the call above allowed for.  This is not an error,
-	 * just an interesting fact.
+	/* Get capabilities for egress kind function, it is not mandatory
+	 * to support the same as the current test suite only tests with
+	 * egress packet IO.
 	 */
-	num_records = MAX_CAPABILITIES;
-	if (MAX_CAPABILITIES < rc)
-		ODPH_DBG("There were more than %u capabilities (%u)\n",
-			 MAX_CAPABILITIES, rc);
-	else
-		num_records = rc;
+	odp_tm_egress_init(&egress);
+	egress.egress_kind = ODP_TM_EGRESS_FN;
+	rc = odp_tm_egress_capabilities(&capabilities_array[1], &egress);
+	if (!rc)
+		num_records++;
```

Comment:
This seems to assume that lack of TM support causes the capability function to fail instead of returning capabilities that indicate no support. That was a bit of a surprise to me. See the comment in the API patch.